### ISSUE TITLE
Unable to parse the `#` symbol for private methods of a class

### ISSUE THREAD

**Issue by iShibi** [vitest-dev/vitest#318]
### Describe the bug

ℹ️ I've opened a related issue at acornjs as requested: https://github.com/acornjs/acorn-class-fields/issues/17

Vitest throws error when testing an instance of a class that implements a private method. The error occurs because private method's name starts with the `#` symbol which it is unable to parse.

The project that I'm trying to add vitest to: https://github.com/twitterjs/twitter.js/pull/96

### Reproduction

https://stackblitz.com/edit/node-stkuzb

### System Info

```shell
System:
  OS: Windows 10 10.0.19043
  CPU: (12) x64 AMD Ryzen 5 4600H with Radeon Graphics
  Memory: 2.09 GB / 7.36 GB
Binaries:
  Node: 16.9.0 - ~\AppData\Local\Volta\tools\image\node\16.9.0\node.EXE      
  Yarn: 1.22.11 - ~\AppData\Local\Volta\tools\image\yarn\1.22.11\bin\yarn.CMD
  npm: 7.24.2 - ~\AppData\Local\Volta\tools\image\npm\7.24.2\bin\npm.CMD     
Browsers:
  Edge: Spartan (44.19041.1266.0), Chromium (96.0.1054.62)
  Internet Explorer: 11.0.19041.1202
npmPackages:
  vitest: ^0.0.113 => 0.0.113
```

### Used Package Manager

npm

### Logs

```shell
FAIL  src/client/__tests__/Client.test.ts [ src/client/__tests__/Client.test.ts ]
Error: Parse failure: Unexpected token (60:8)
Contents of line 60:   async #fetchClientUser(username) {
 ❯ ssrTransform node_modules/vite/dist/node/chunks/dep-fcec4469.js:56614:15
 ❯ doTransform node_modules/vite/dist/node/chunks/dep-fcec4469.js:57940:48
 ❯ processTicksAndRejections ../../../node:internal/process/task_queues:96:5
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by iShibi** [comment]
from the above mentioned issue:
> As of Acorn 8.2.0 private fields and private methods are supported in Acorn without the need to use this (or another) plugin, I recommend upgrading Acorn.

**Comment by sheremet-va** [comment]
Should be fixed in [0.0.120](https://github.com/vitest-dev/vitest/releases/tag/v0.0.120)
